consul/consul.go

```diff
@@ -80,12 +80,18 @@ func (c *Consul) getServicesUsingAgent(name string, agent *consulAPI.Client) ([]
 	return allServices, nil
 }
 
-func dcAwareQueriesForAllDCs(agent *consulAPI.Client) ([]*consulAPI.QueryOptions, error) {
+func dcAwareQueriesForAllDCs(agent *consulAPI.Client, dc string) ([]*consulAPI.QueryOptions, error) {
+	if dc != "" {
```

Comment:
This `dc` parameter shares a name with a variable on line 96 - this could lead to confusion, make it more unique perhaps?

Reply:
fixd